VERSIONING.md

```diff
@@ -0,0 +1,220 @@
+          whenever you are using the module name).
+        * If a module is version `v0` or `v1`, do not include the major version
+          in either the module path or the import path.
+    * In addition to public APIs, telemetry produced by stable instrumentation
+      will remain stable and backwards compatible. This is to avoid breaking
+      alerts and dashboard.
+    * Modules will be used to encapsulate receivers, processor, exporters,
+      extensions, and any other independent sets of related components.
+        * Experimental modules still under active development will be versioned at
+          `v0` to imply the stability guarantee defined by
+          [semver](https://semver.org/spec/v2.0.0.html#spec-item-4).
+
+          > Major version zero (0.y.z) is for initial development. Anything MAY
+          > change at any time. The public API SHOULD NOT be considered stable.
+
+        * Mature modules for which we guarantee a stable public API and telemetry will
+          be versioned with a major version greater than `v0`.
+        * Experimental modules will start their versioning at `v0.0.0` and will
+          increment their minor version when backwards incompatible changes are
+          released and increment their patch version when backwards compatible
+          changes are released.
+        * Stable contrib modules cannot depend on experimental modules from this
+          project.
+        * All stable contrib modules of the same major version with this project
+          will use the same entire version as this project.
+            * Stable modules may be released with an incremented minor or patch
+              version even though that module's code has not been changed. Instead
+              the only change that will have been included is to have updated that
+              modules dependency on this project's stable APIs.
+            * When an experimental module in contrib becomes stable a new stable
+              module version will be released and will include this now stable
+              module. The new stable module version will be an increment of the minor
+              version number and will be applied to all existing stable contrib
+              modules, this project's modules, and the newly stable module being
+              released.
+    * Contrib modules will be kept up to date with this project's releases.
+        * Due to the dependency contrib modules will implicitly have on this
+          project's modules the release of stable contrib modules to match the
+          released version number will be staggered after this project's release.
+          There is no explicit time guarantee for how long after this projects
+          release the contrib release will be. Effort should be made to keep them
+          as close in time as possible.
+        * No additional stable release in this project can be made until the
+          contrib repository has a matching stable release.
```

Comment:
Is this a necessary requirement? What are we trying to achieve by this? Why do we want to prevent an urgent core release (e.g. security related) if contrib is lagging?

Reply:
I have removed the sub-bullets here, keeping only the general policy statement that contrib modules will be kept up to date without discussion of specific mechanisms.